neuralop/data/transforms/rescale.py

```diff
@@ -0,0 +1,372 @@
+import torch
+import numpy as np
+import torchvision.transforms.functional as F
+
+
+def DarcyExtractBC(x, y):
+    """
+    Extract boundary conditions for Darcy Equation: 2D boundary value problem.
+    Code adapted from paper https://arxiv.org/abs/2507.18813
+
+    Attributes
+    ----------
+    x: torch.Tensor
+        initial condition, a tensor of shape (batch, channels, X, Y)
+    y: torch.Tensor
+        boundary condition, a tensor of shape (batch, channels, X, Y)
+
+    Returns
+    -------
+    x: torch.Tensor
+        the combined input tensor with boundary conditions added in channels 1,...,4
+    """
+    unsqueeze_x = x.ndim == 3  # check whether batch dimension is missing
```

Comment:
Perhaps instead of inferring the batch dynamically, it would be simpler to have the forward call `DarcyExtractBC(x,y,batched=True)`?